Lib/idlelib/ParenClose.py

```diff
@@ -0,0 +1,91 @@
+"""
+Parens and Ticks Closing Extension
+
+When you hit left paren or tick,
+automatically creates the closing paren or tick.
+
+Author: Charles M. Wohlganger
+        charles.wohlganger@gmail.com
+
+Last Updated: 12-Aug-2017 by Charles Wohlganger
+
+Add to the end of config-extensions.def :
+
+    [ParenClose]
+    enable = True
+    paren_close = True
+    tick_close = True
+    skip_closures = True
+    comment_space = True
+    [ParenClose_cfgBindings]
+    p-open = <Key-parenleft> <Key-bracketleft> <Key-braceleft>
+    t-open = <Key-'> <Key-">
+    p-close = <Key-parenright> <Key-bracketright> <Key-braceright>
+
+"""
+
+from idlelib.config import idleConf
+
+
+class ParenClose:
+    '''
+    When loaded as an extension to IDLE, and paren_close is True, the symbols
+    ([{ will have their closures printed after them and the insertion cursor
+    moved between the two.  The same is true for tick closures and the symbols
+    ' and ".  When \'\'\' or """ are typed and tick_close is True, it will also
+    produce the closing symbols. If skip_closures is True, then when a closure
+    symbol is typed and the same one is to the right of it, that symbols is
+    deleted before the new one is typed, effectively skipping over the closure.
+    '''
+    def __init__(self, editwin=None): #setting default to none makes testing easier
```

Comment:
Inline comments should be written like a sentence - begin with a capital letter and end with a period.